sale_order_general_discount/models/product_product.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2021 - Pierre Verkest
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    general_discount_apply = fields.Boolean(
+        string="Apply general discount",
+        default=True,
+        required=True,
```

Comment:
why having a boolean with a default value as required?

Reply:
well as far odoo do not manage properly Null database value this as no real impact, not sure it changed things while installing this module. Do you want me to remove it ?

Reply:
thanks @petrus-v, it was more out of curiosity :)

Reply:
ok ! thanks for review, I'll rebase this PR to fix conflicts!...